contracts/lens/OpinionatedOracle.sol

```diff
@@ -0,0 +1,102 @@
+pragma solidity =0.7.6;
+
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Factory.sol';
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '../libraries/WeightedOracleLibrary.sol';
+import '../libraries/OracleLibrary.sol';
+
+contract OpinionatedOracle {
+    IUniswapV3Factory factory;
+
+    enum ManipulationResistance {
+        Dangerous, // Spot price from beginning of the block
```

Comment:
i wish we could do `Dangerous = 0 seconds, Weak = 60 seconds, ...` like in typescript